rust/relay/src/server.rs

```diff
@@ -172,13 +176,28 @@ where
             data_relayed_counter.clone(),
         );
 
+        // Although not explicitly recommended by the RFC, we allow the user to pass a custom range.
+        // This is helpful when debugging, running NATed (such as in Docker), or when deploying
+        // in production where the default range conflicts with other services.
+        let lowest_port = std::env::var("TURN_LOWEST_PORT")
+            .ok()
+            .and_then(|s| s.parse().ok() as Option<u16>)
+            .unwrap_or(RFC_LOWEST_PORT);
+        let highest_port = std::env::var("TURN_HIGHEST_PORT")
+            .ok()
+            .and_then(|s| s.parse().ok() as Option<u16>)
+            .unwrap_or(RFC_HIGHEST_PORT);
```

Comment:
Instead of doing this here, I'd add two fields to `Args` in `main.rs` and pass the values through.

Not only does clap supports default values (see `portal_ws_url`) with automatic documentation in `--help` but this `Server` is also meant to be sans-IO, meaning it should pure in a functional programming sense. Reading from the environment here would make it difficult to run parallel tests for example.

Reply:
Ah, very nice. Rust is great.